### PR TITLE
Change Delivery Request Body for Notifications

### PR DESCRIPTION
**Pull request by davidcui-amzn**
### Description
* Change the request body for report definitions for Notifications integration in OpenSearch and OpenSearch Dashboards
Sample request body:
```
"delivery": {
  confidIds: ["id1", "id2"],
  textDescription: "textDescription",
  htmlDescription: "htmlDescription",
  title: "title"
}
```
* Update references in Report definition details to prevent error toast on visiting page (further UI changes to come)
* Update all test cases

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-reports/server/model/index.ts [hunk 1]
```diff
@@ -187,16 +187,10 @@ export const triggerSchema = schema.object({
...
-    schema.literal(DELIVERY_TYPE.opensearchDashboardsUser),
-    schema.literal(DELIVERY_TYPE.channel),
-  ]),
-  delivery_params: schema.conditional(
-    schema.siblingRef('delivery_type'),
-    DELIVERY_TYPE.opensearchDashboardsUser,
-    opensearchDashboardsUserSchema,
-    channelSchema
-  ),
+  configIds: schema.maybe(schema.arrayOf(schema.string())),
```
Comment (zhongnansu) [inline dashboards-reports/server/model/index.ts:190]: why `maybe()`, isn't this a required field?

**Review by zhongnansu — APPROVED** [review]
lgtm, left one comment

File: dashboards-reports/server/model/index.ts (hunk 1 above)
Comment (davidcui-amzn) [reply dashboards-reports/server/model/index.ts:190]: Yep, good catch. Will fix that

**Comment by zhongnansu** [comment]
Will current report fail after you change the model/input validation? Any temp placeholder you set to pass the IT(cypress test)

**Comment by davidcui-amzn** [comment]
> Will current report fail after you change the model/input validation? Any temp placeholder you set to pass the IT(cypress test)

It shouldn't fail but I will run locally before pushing my tests. The placeholder is the empty array but the `configIds` field always exists

**Review by joshuali925 — APPROVED** [review]
lgtm

**Pull request merged**